extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,491 @@
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/common/types/blob.hpp"
+#include "duckdb/main/extension_util.hpp"
+#include "inet_extension.hpp"
+#include "inet_functions.hpp"
+#include "html_charref.hpp"
+
+namespace duckdb {
+
+struct UnescapeBindData : public FunctionData {
+	const HTML5NameCharrefs html5_names;
+	const unordered_set<uint32_t> invalid_codepoints;
+	const unordered_map<uint32_t, const char *> special_invalid_charrefs;
+
+	UnescapeBindData()
+	    : html5_names(HTML5NameCharrefs()), invalid_codepoints(ReturnInvalidCodepoints()),
+	      special_invalid_charrefs(ReturnInvalidCharrefs()) {
+	}
+
+	unique_ptr<FunctionData> Copy() const override {
+		return make_uniq<UnescapeBindData>();
+	}
+
+	bool Equals(const FunctionData &other_p) const override {
+		auto &other = other_p.Cast<UnescapeBindData>();
+		bool equal_sizes = (html5_names.mapped_strings.size() == other.html5_names.mapped_strings.size() &&
+		                    invalid_codepoints.size() == other.invalid_codepoints.size() &&
+		                    special_invalid_charrefs.size() == other.special_invalid_charrefs.size());
+		return (equal_sizes && (html5_names.mapped_strings == other.html5_names.mapped_strings &&
+		                        invalid_codepoints == other.invalid_codepoints &&
+		                        special_invalid_charrefs == other.special_invalid_charrefs));
+	}
+};
+
+unique_ptr<FunctionData> UnescapeBind(ClientContext &context, ScalarFunction &bound_function,
+                                      vector<unique_ptr<Expression>> &arguments) {
+	return make_uniq<UnescapeBindData>();
+}
+
+static idx_t to_utf8(uint32_t code_point, char arr[4]) {
```

Comment:
I feel like we have this function already, in `utf8proc.hpp` as `utf8proc_encode_char`
Is this different somehow?